libcxx/test/std/utilities/memory/util.smartptr/util.smartptr.shared/util.smartptr.shared.create/make_shared_for_overwrite.pass.cpp

```diff
@@ -60,6 +60,9 @@ constexpr char pattern = 0xDE;
 
 void* operator new(std::size_t count) {
   void* ptr = std::malloc(count);
+  if (!ptr) {
+    std::abort(); // placate MSVC's unchecked malloc warning
```

Comment:
I like the comment, but when not an assert instead?

Reply:
Unfortunately, static analysis doesn't consider `assert`'s effect on control flow. I'm not sure if this is a compiler bug or by design (because `assert` is controlled by `NDEBUG`). That's why the codebase currently uses `if-abort`.